pkg/provision/mbus/mbus.go

```diff
@@ -0,0 +1,61 @@
+package mbus
+
+import (
+	"context"
+
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/zos/pkg/mbus"
+	"github.com/threefoldtech/zos/pkg/provision"
+)
+
+const (
+	cmdDeploy = "msgbus.deploy"
+	cmdDelete = "msgbus.delete"
+)
+
+// WorkloadsMessagebus is provision engine Workloads
+type WorkloadsMessagebus struct {
+	engine provision.Engine
+	mbus   *mbus.Messagebus
+}
+
+// NewWorkloadsMessagebus creates a new messagebus instance
+func NewWorkloadsMessagebus(engine provision.Engine, port uint16) (*WorkloadsMessagebus, error) {
+	messageBus, err := mbus.New(port, context.Background())
+	if err != nil {
+		return nil, err
+	}
+
+	api := &WorkloadsMessagebus{
+		engine: engine,
+		mbus:   messageBus,
+	}
+
+	return api, nil
+}
+
+func (w *WorkloadsMessagebus) Run() error {
```

Comment:
I don't like many things about this method
- it (again) requires knowledge of the internal of the message bus, the "Message" structure, the streams, and so on.
- Also a call to StreamMessages is already a "blocking" call, it will work in an infinite loop to pull messages (which is fine) but what is not fine, that the 2nd call to StreamMessages will never happen because the first call will never return.
- Why do i have to define channels, and do select, and much much more things that can be optimized and abstracted by the library itself.
- For example, and HTTP library, doesn't require knowledge of how the headers or the http request parsing is happening, while u as a user can still "get" this information from the request object. 

In my opinion, the library should have the following Interface:
```go

bus := rmb.New(url) // url to redis endpoint

bus.Handle("zos.deployment.deploy", deployHandler)
bus.Handle("zos.deployment.get", getHandler)
bus.Handle("zos.deployment.delete", deleteHandler)

sub := bus.Subrouter("network")
sub.Handle("config.get") // this should be similar to `bus.Handle("network.config.get")` but is cleaner to define may be and allow later injection of middle ware for different parts

// finally
bus.Run()
```

a function then can be something like
```go
func getHandler(ctx context.Context, payload bytes[]) (interface{}, error) {
  // load your pay load to parse request params
  // do your things
 // if needed access extra messages fields from ctx, like
 message := rmb.GetMessage(ctx)
  // or others!
  return result, nil
}
```